pkg/operation/botanist/secrets.go

```diff
@@ -350,9 +350,11 @@ func (b *Botanist) rotateSSHKeypairSecrets(ctx context.Context, gardenerResource
 	}
 	gardenerResourceDataList.Delete(v1beta1constants.SecretNameSSHKeyPair)
 
-	// remove operation annotation
+	// remove operation annotation and annotate shoot with GardenerSSHRotation annotation indicating that Shoot's ssh-keypair
+	// has been rotated.
 	return b.Shoot.UpdateInfo(ctx, b.K8sGardenClient.Client(), false, func(shoot *gardencorev1beta1.Shoot) error {
 		delete(shoot.Annotations, v1beta1constants.GardenerOperation)
+		metav1.SetMetaDataAnnotation(&shoot.ObjectMeta, v1beta1constants.GardenerSSHRotation, v1beta1constants.ShootOperationSSHKeypairRotated)
```

Comment:
Instead of storing this annotation on the `Shoot`, can we just ensure that the `ssh-keypair.old` secret in the shoot namespace in the seed only exists when it was really rotated at least once? Then we can check if this secret exists and only sync to garden if this condition holds true?

Reply:
How can we check whether the `ssh-keypair` is rotated atleast once , as the annotation `gardener.cloud/operation : rotate-ssh-keypair` gets removed after reconcilation of shoot ?

Reply:
> can we just ensure that the `ssh-keypair.old` secret in the shoot namespace in the seed only exists when it was really rotated at least once? Then we can check if this secret exists and only sync to garden if this condition holds true?

Reply:
Addressed the feedback in commit [267ef07](https://github.com/gardener/gardener/pull/5388/commits/267ef07500aadb48572e04da0637038a304ed92d).